compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2376,11 +2376,14 @@ private boolean checkFieldEquivalency(BRecordType lhsType, BRecordType rhsType,
         for (BField lhsField : lhsType.fields.values()) {
             BField rhsField = rhsFields.get(lhsField.name.value);
 
-            // There should be a corresponding RHS field
+            // If LHS field is required, there should be a corresponding RHS field
             if (rhsField == null) {
-                return false;
+                if (Symbols.isOptional(lhsField.symbol)) {
+                    continue;
+                } else {
+                    return false;
+                }
```

Comment:
Can't we rearrange this, 
```
if (!Symbols.isOptional(lhsField.symbol)) {
                     return false;
}
```

Reply:
I think we cannot do this. Because when `rhsField == null` and `Symbols.isOptional(lhsField.symbol)` becomes `true`, then it check for flags also.(L2661).

Reply:
We can remove the else though
```suggestion
                if (!Symbols.isOptional(lhsField.symbol)) {
                    return false;
                }
                continue;
```

Reply:
Changed.